### ISSUE TITLE
Add support for multiple-type derivation aliases

### ISSUE THREAD

**Issue by ivan-klass** [tofu-tf/derevo#133]
It would be good to have an ability to alias mutliple derivations at once, so instead of etc
```scala
import com.sksamuel.avro4s.{Encoder, Decoder, SchemaFor}

@derive(Encoder, Decoder, SchemaFor)
case class FooBar(foo: String, bar: Int)
```
one can write (preferred)
```scala
@avro4sCodec
case class FooBar(foo: String, bar: Int)
```

or
```scala
@derive(avro4sCodec)
case class FooBar(foo: String, bar: Int)
```

**Comment by ivan-klass** [comment]
Some thoughts:

For example, the alias could be implemented as 
```scala
import derevo.MultipleAlias
class avro4sCodec extends MultipleAlias(Encoder, Decoder, SchemaFor)
```
or with type-builders like shapeless HList but for HK:
```scala
sealed trait Many[T <: Many[_]]
trait T extends Many[Nothing]
case class ::[A[_], Tail <: Many[_]]() extends Many[Tail]

class deriveMany[T <: Many[_]]() extends scala.annotation.Annotation

object example {
  type TypeClasses = List :: Option :: Vector :: T
  class alias extends deriveMany[example.TypeClasses]
}

// all forms are valid
@example.alias
@deriveMany[example.TypeClasses]
@deriveMany[Either[String, *] :: Function[*, Int] :: T]
case class FooBar(int: Int, str: String)


println("compiles")
```
the latter [compiles at scastie](https://scastie.scala-lang.org/gNKbIRLPThmUufYR3G3PfQ)

**Comment by zawodskoj** [comment]
Implemented in #333
